packages/components/src/internal/components/forms/input/RadioGroupInput.tsx

```diff
@@ -86,7 +87,12 @@ class RadioGroupInputImpl extends PureComponent<RadioGroupInputProps, State> {
                                 onChange={this.onValueChange}
                                 disabled={option.disabled}
                             />
-                            <span className={classNames('radioinput-label', { selected })}>{option.label}</span>
+                            <span
+                                className={classNames('radioinput-label', { selected })}
+                                onClick={() => this.onSetValue(option.value)}
```

Comment:
This should be memoized, you could do this by creating a `RadioInputOption` component that memoizes the `onChange` callback.

Reply:
done